manifests/prepare.pp

```diff
@@ -51,14 +53,87 @@
     }
   }
 
-  # TODO: manage server.cfg contents
-  if $::mcollective_configured {
-    file { '/etc/puppetlabs/mcollective':
-      ensure => directory,
+  # manage client.cfg and server.cfg contents
+  file { '/etc/puppetlabs/mcollective':
+    ensure => directory,
+  }
+  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
+  if $::mco_server_config {
+    $mco_server = '/etc/puppetlabs/mcollective/server.cfg'
+    file { $mco_server:
+      ensure  => file,
+      source  => $::mco_server_config,
+    }
+
+    if $::mco_server_settings {
+      $mco_server_libdir = $::mco_server_settings['libdir']
+      if $mco_server_libdir {
+        ini_setting { 'server/libdir':
+          section => '',
+          setting => 'libdir',
+          path    => $mco_server,
+          value   => "${mco_server_libdir}:/opt/puppetlabs/mcollective/plugins",
```

Comment:
The resolution of the $mco_server_settings fact parses a config file based on the value of the $mco_server_config fact, which is noted as having this behaviour:

```
# The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS
```

Which would grab the AIO settings again, as PE and AIO paths for server.cfg are the same - `/etc/puppetlabs/mcollective/server.cfg`

That said, looking closer the implementation of $mco_server_config is actually going to pick FOSS over the PE/AIO path when both files exist, so I guess it might work out when migrating a FOSS install.